Cython/Utility/ExtensionTypes.c

```diff
@@ -1,3 +1,65 @@
+/////////////// PyType_Ready.proto ///////////////
+
+static int __Pyx_PyType_Ready(PyTypeObject *t);
+
+/////////////// PyType_Ready ///////////////
+
+// Wrapper around PyType_Ready() with some runtime checks and fixes
+// to deal with multiple inheritance.
+static int __Pyx_PyType_Ready(PyTypeObject *t) {
+    /* Loop over all bases (except the first) and check that those
+     * really are heap types. Otherwise, it would not be safe to
+     * subclass them.
+     *
+     * We also check tp_dictoffset: it is unsafe to inherit
+     * tp_dictoffset from a base class because the object structures
+     * would not be compatible. So, if our extension type doesn't set
+     * tp_dictoffset (i.e. there is no __dict__ attribute in the object
+     * structure), we need to check that none of the base classes sets
+     * it either. */
```

Comment:
Could you please use line comments here? Multi-line C comments do not get discarded, and there is not much use in copying this into the generated .c files.